src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -56,22 +56,41 @@ public void storeMessage(Message message) {
    */
   public List<Message> getMessages(String recipient) {
     List<Message> messages = new ArrayList<>();
-
     Query query =
         new Query("Message")
             .setFilter(new Query.FilterPredicate("recipient", FilterOperator.EQUAL, recipient))
             .addSort("timestamp", SortDirection.DESCENDING);
-    PreparedQuery results = datastore.prepare(query);
+    messages = message(query);
+    }
+
+    return messages;
+  }
+  public List<Message> getAllMessages(){
+    List<Message> messages = new ArrayList<>();
+   
+    Query query = new Query("Message")
+      .addSort("timestamp", SortDirection.DESCENDING);
 
+    for (Entity entity : results.asIterable()) {
+      try {
+        messages = message(query);  
+      } catch (Exception e) {
+        System.err.println("Error getting messages.");
+        System.err.println(entity.toString());
+        e.printStackTrace();
+      }
+    }
```

Comment:
It looks like this entire for loop (including the try-catch) is left over from previous implementation and is already executed by the helper method you implemented (`message(Query query)`) you can remove it and simply keep `messages = message(query);`. Please let me know if that doesn't seem to make sense.